dali/pipeline/data/buffer.h

```diff
@@ -464,8 +546,9 @@ class DLL_PUBLIC Buffer {
   Index size_ = 0;                   // The number of elements in the buffer
   size_t num_bytes_ = 0;             // To keep track of the true size of the underlying allocation
   int device_ = CPU_ONLY_DEVICE_ID;  // device the buffer was allocated on
+  AccessOrder order_;                // The order of memory access (host or device)
   bool shares_data_ = false;         // Whether we aren't using our own allocation
-  bool pinned_ = !RestrictPinnedMemUsage();  // Whether the allocation uses pinned memory
+  bool pinned_ = false;              // Whether the allocation uses pinned memory
```

Comment:
Maybe we should do this in a separate PR?

Reply:
I'll revert and see how it goes...

Reply:
I'm not saying it is wrong. But it is a big change and maybe it is better to add it separately.